### DISCUSSION TITLE
HomeBridge Plugin Not Detecting PetSafe Feeder

### DISCUSSION THREAD

**Discussion by walkerj001** [homebridge-plugins/homebridge-petsafe-smart-feed#48] [General]
Hello, I was super excited to see this plugin and support for homebridge. I attempted to install several ways and it will not detect my PetSafe feeder no matter what I do. Any help would be very much appreciated. Token was generated and I got email with code no problem. Upon restart homebridge does not list any feeder. "[PetSafeSmartFeed] Configuring 0 PetSafe Smart Feeder(s):   " I would be happy to donate to this project if we can get an update to resolve issues!! Thanks

**Comment by dgreif** [comment]
Weird, haven't had any other reports of it not working. Which model of the smart feeder do you have? An Amazon link would probably be helpful if you have it

**Reply by walkerj001** [reply]
Hello,

Here is the link of the smart feeder I have. It’s a Gen 2 Feeder and the originally one I got was defective after a few months. The company then sent me a new one and stated it was same one . New feeder is showing firmware V2.0.9 

Amazon Link: PetSafe Smart Feed Automatic Dog and Cat Feeder - Smartphone - Wi-Fi Enabled for iPhone and Android Smartphones https://www.amazon.com/dp/B073WYP317/ref=cm_sw_r_cp_api_i_D41683X62MQX46YZ2VPT?_encoding=UTF8&psc=1

**Reply by dgreif** [reply]
The link you provided appears to be for the Gen 1.  Here is the Gen 2: https://www.amazon.com/PetSafe-Automatic-Generation-Smartphone-Replenishment-dp-B07NR47N2Q/dp/B07NR47N2Q/ref=dp_ob_title_pet?th=1.  I honestly can't say what the difference is between the two.  I would think the API would be similar between the two, but it appears _something_ is different, which is why it's not working with the homebridge plugin.  You could eithe exchange it for the one I linked, or I could take a look at the API if you were willing to share your credentials with me (via email, not on GitHub).  No promises on when I would get around to it if you choose the second option though 😅

**Reply by walkerj001** [reply]
I have two of these feeders so I’ll get my second one online and login with an account and provide you the info. I’ll get it up and online soon so you can look at API. Other one is used a lot, odd part to me is I looked up what I purchased vs your link and they are both Gen 2 units with Amazon dash and Alexa ability. I don’t honestly know the difference even after looking for 20 mins. I’ll have to do some more digging. What’s an email I can send info to ? Also a huge bonus if slow feed time can be adjusted. It was originally the only reason I got it. Little did I know it’s the worse programming from the company. It drops a couple pebbles then dog has to sit and wait a few. That slow feed time would be way better if adjustable. Getting it to work first would be progress. I really appreciate your help with this!

**Reply by walkerj001** [reply]
Just an update I originally purchased the Gen 2 unit (the link you sent) and I received wrong one it appeared. They are sending me the correct unit and I will retry later this week with the Gen 2 PetSafe feeder. Hopefully it will work with my homebridge setup then. Be a miracle if in your future updates where slow feed time could be adjusted. Thanks again for everything you done for this project. Super excited to get this going!

**Reply by dgreif** [reply]
Glad you got that part sorted out @walkerj001!  As far as slow feeds go, I think it's just an on/off setting with the UI calls so I wouldn't be able to adjust it.

**Reply by walkerj001** [reply]
I should be getting the new Gen2 feeder tomorrow or Friday so we will see! I understand on the slow feed setting just wasn't sure. Amazon stated it was a set variable designed to dispense the food over 1/8 cup over 15 minutes of time. I was hoping maybe it was adjustable in the programming. If not completely understand!! I will update you when I get new feeder and get it hooked up. Thanks for all the help , greatly appreciated. @dgreif

**Reply by walkerj001** [reply]
Got new feeder tonight and is now detected and working. Thanks! @dgreif

**Comment by dallowbg** [comment]
Is it ok if I drop into this discussion?
I have the Gen 1 version as mentioned in the conversation and did my setup today. I see in the logs that it does configure 1 Smartfeed device: "[PetSafeSmartFeed] Configuring 1 PetSafe Smart Feeder(s):" whereas the original poster showed 0.

However, am I supposed to see the device in Homekit after that? I looked but can't seem to find it, even after restarting Homebridge.

**Reply by dgreif** [reply]
Hmm that's interesting.  Do you have other homebridge devices/plugins set up?  How are you looking for the device? (homebridge ui vs Home app)

**Reply by dallowbg** [reply]
Hi dgreif.
The issue resolved itself actually!
Originally, I could see it in the Homebridge UI under accessories but it did not appear in the Home app at all).
Last week it randomly popped up in the Home app finally and seems to work fine (meal triggered when I activate switch).

Thanks!

**Comment by rickjtaylor** [comment]
I have a Gen 1 feeder and am also getting “Configuring 0 PetSafe Smart Feeder(s)”. I would be happy to help troubleshoot to see if we can that working. I have had it for awhile and don’t have the ability to exchange it.

**Reply by dgreif** [reply]
@rickjtaylor I haven't had a lot of time to work on this plugin lately.  If you are able to dive into it yourself and open a PR, I'd be happy to take a look.  Otherwise, I'm not sure I'll get around to adding Gen 1 support any time soon.
